### PR TITLE
Floor Goblin tweaks

### PR DESCRIPTION
**Pull request by PrimeNumb**
## About the PR 
- Fixes an issue where Ankle Bite would ignore range (cooldown also modified)
- Fixes an issue where abilities were able to be cast while incapacitated
- Steal Shoes no longer tries to cast if the target has no shoes

## Why's this needed? 
Refining the gimmick

### CODE REVIEW

File: code/datums/abilities/diabolical.dm [hunk 1]
```diff
@@ -288,6 +288,13 @@
 	targeted = 0
 	cooldown = 0
 
+	tryCast()
+		if (isunconscious(holder.owner) || holder.owner.hasStatus("weakened"))
```
Comment (ZeWaka) [inline code/datums/abilities/diabolical.dm:292]: Don't you mean `is_incapacitated`?

File: code/datums/abilities/diabolical.dm [hunk 2]
```diff
@@ -288,6 +288,13 @@
...
+			boutput(holder.owner, "<span class='alert'>You cannot cast this ability while you are incapacitated.</span>")
+			src.holder.locked = 0
+			return 999
+		return ..()
```
Comment (ZeWaka) [inline code/datums/abilities/diabolical.dm:296]:
```suggestion
		. = ..()
```

File: code/modules/antagonists/floor_goblin/floor_goblin.dm [hunk 3]
```diff
@@ -82,10 +82,17 @@
 	desc = "Trip a target by biting at their ankles."
 	icon = 'icons/mob/critter_ui.dmi'
 	icon_state = "clown_spider_bite"
-	cooldown = 10 SECONDS
+	cooldown = 20 SECONDS
 	targeted = 1
 	target_anything = 1
 
+	tryCast()
+		if (isunconscious(holder.owner) || holder.owner.hasStatus("weakened"))
```
Comment (ZeWaka) [inline code/modules/antagonists/floor_goblin/floor_goblin.dm:90]: see: incap

File: code/modules/antagonists/floor_goblin/floor_goblin.dm [hunk 4]
```diff
@@ -82,10 +82,17 @@
...
+			boutput(holder.owner, "<span class='alert'>You cannot cast this ability while you are incapacitated.</span>")
+			src.holder.locked = 0
+			return 999
+		return ..()
```
Comment (ZeWaka) [inline code/modules/antagonists/floor_goblin/floor_goblin.dm:94]:
```suggestion
		. = ..()
```

File: code/modules/antagonists/floor_goblin/floor_goblin.dm [hunk 5]
```diff
@@ -135,6 +142,13 @@
 	targeted = 1
 	target_anything = 1
 
+	tryCast()
+		if (isunconscious(holder.owner) || holder.owner.hasStatus("weakened"))
```
Comment (ZeWaka) [inline code/modules/antagonists/floor_goblin/floor_goblin.dm:146]: see: incap

File: code/modules/antagonists/floor_goblin/floor_goblin.dm [hunk 6]
```diff
@@ -135,6 +142,13 @@
...
+			boutput(holder.owner, "<span class='alert'>You cannot cast this ability while you are incapacitated.</span>")
+			src.holder.locked = 0
+			return 999
+		return ..()
```
Comment (ZeWaka) [inline code/modules/antagonists/floor_goblin/floor_goblin.dm:150]:
```suggestion
		. = ..()
```

**Review by ZeWaka — APPROVED** [review]

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**